src/Tribe/RSVP.php

```diff
@@ -915,139 +876,94 @@ public function create_attendee_for_ticket( $ticket, $attendee_data ) {
+		$attendee_full_name = empty( $attendee_data['full_name'] ) ? null : sanitize_text_field( $attendee_data['full_name'] );
+		$attendee_optout    = empty( $attendee_data['optout'] ) ? 0 : (int) tribe_is_truthy( $attendee_data['optout'] );
 
-		if ( empty( $data['order_status'] ) || ! $this->tickets_view->is_valid_rsvp_option( $data['order_status'] ) ) {
-			$attendee_order_status = null;
+		if ( empty( $attendee_data['order_status'] ) || ! $this->tickets_view->is_valid_rsvp_option( $attendee_data['order_status'] ) ) {
+			$attendee_status = null;
 		} else {
-			$attendee_order_status = $data['order_status'];
+			$attendee_status = $attendee_data['order_status'];
 		}
 
-		$product_id  = $attendee['product_id'];
+		$ticket_id  = $attendee['product_id'];
 
 		//check if changing status will cause rsvp to go over capacity
-		$previous_order_status = get_post_meta( $order_id, self::ATTENDEE_RSVP_KEY, true );
+		$previous_order_status = get_post_meta( $attendee_id, self::ATTENDEE_RSVP_KEY, true );
 
 		// The status changed from "not going" to "going", check if we have the capacity to support it.
-		if ( tribe_is_truthy( $attendee_order_status ) && in_array( $previous_order_status, $this->get_statuses_by_action( 'count_not_going' ), true ) ) {
-			$capacity = tribe_tickets_get_capacity( $product_id );
-			$sales = (int) get_post_meta( $product_id, 'total_sales', true );
-			$unlimited = -1;
+		if ( tribe_is_truthy( $attendee_status ) && in_array( $previous_order_status, $this->get_statuses_by_action( 'count_not_going' ), true ) ) {
+			$capacity  = tribe_tickets_get_capacity( $ticket_id );
+			$sales     = (int) get_post_meta( $ticket_id, 'total_sales', true );
+			$unlimited = - 1;
+
+			// If capacity is not unlimited and there is no capacity for switching, remove attendee status from being saved.
 			if ( $unlimited !== $capacity && $sales + 1 > $capacity ) {
-				return;
+				// Set to null so it won't be changed.
+				$attendee_status = null;
 			}
 		}
 
-		$this->update_sales_and_stock_by_order_status( $order_id, $attendee_order_status, $product_id );
-
-		if ( null !== $attendee_order_status ) {
-			update_post_meta( $order_id, self::ATTENDEE_RSVP_KEY, $attendee_order_status );
-		}
-
-		update_post_meta( $order_id, self::ATTENDEE_OPTOUT_KEY, $attendee_optout );
+		$this->update_sales_and_stock_by_order_status( $attendee_id, $attendee_status, $ticket_id );
 
-		if ( null !== $attendee_full_name ) {
-			update_post_meta( $order_id, $this->full_name, $attendee_full_name );
-		}
+		$attendee_data_to_save = [
+			'full_name'       => $attendee_full_name,
+			'email'           => $attendee_email,
+			'attendee_status' => $attendee_status,
+			'optout'          => $attendee_optout,
+		];
 
-		if ( null !== $attendee_email ) {
-			update_post_meta( $order_id, $this->email, $attendee_email );
-		}
+		// Remove arguments that are null.
+		$attendee_data_to_save = array_filter( $attendee_data_to_save, static function ( $value ) {
```

Comment:
* :no_entry_sign: Opening parenthesis of a multi-line function call must be the last content on the line
* :no_entry_sign: Only one argument is allowed per line in a multi-line function call

<sub>hash: d3938f645509c485ff6ffc964e79ee7d</sub>